### ISSUE TITLE
Project style 

### ISSUE THREAD

**Issue by Kishy-nivas** [kashyapan/Playing-with-SFML#1]
Try to avoid  "using namespace sf "  for small programs it is ok , once we go to make cool projects it is hard to keep track of them,  and avoid having everything in single main func() , 
 big projects require you to use #pragma once or other include guards , try them ,

**Comment by Kishy-nivas** [comment]
If possible try doing a small project , i.e a linked list , a user enters a number that number is pushed as a link and try visually implementing the delete function with the shape rectangle, it will be cool!

**Comment by kashyapan** [comment]
I'm aware of consequences of  using `namespace`  .
I wont be using them once we do projects.

I saw your project they were nicely split into functions and classes.
I'm yet to get familiarize with SFML once I'm comfortable I will be using classes and stuff.

Linked List Sounds cool. I think I can do it now.

**Comment by Kishy-nivas** [comment]
So How is game development so far ? 

To be honest for me it is hard,  it's like 4 times harder than web development, that's what makes it cool I guess. 

The difficulty will be same even with Unreal or any engine  so this sfml provides a nice basement in game development.

**Comment by Kishy-nivas** [comment]
As far I see we are learning a API , with c++ it is a huge task, it  can make us  good programmers, with good design skills , we can easily learn any other famous frameworks like Qt which will be useful for system programs , so learning SFML deeply won't harm us in any way .
 
It will make us good in web development too, which is fairly easy compared to SFML or Unreal .

And only thing is the time, but I think SFML itself is enough to build standalone 2d games, so I think we may need to spend at least a decent 1.5 -2 months (or even more ) with it before moving to Unreal .

In the mean time  we must also  learn linear algebra, 2d-3d physics stuffs, geometry and implement them in SFML itself .
Considering the above stuffs,  even SFML may take more of our time it seems , maybe we must postpone Unreal and concentrate on SFML and Web development side by side before we work with Unreal . 

So what's your view about this problem .

**Comment by kashyapan** [comment]
I have not so far faced much difficulty with SFML (documentation is great).
This may be because I have not seen difficult part yet.

If SFML is taking more time for me so be it. I just compare what I am today as compared to yesterday.

I agree we must master SFML before moving to Unreal Engine (2 - 3 months is very reasonable). There is no point rushing fundamentals. 

> In the mean time we must also learn linear algebra, 2d-3d physics stuffs, geometry and implement them in SFML itself .

I couldn't agree more. I had a glance on MIT OCW linear algebra yesterday (casually) it's taught my great guy. Linear algebra is of great use in ML too so It is cool.

If I get any great material I'll share. I ask you the same

In short : 
Investing more time in SFML  (totally cool)
Learning linear algebra , Geometry (Cool Let's start)
